src/Signer.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Data.h"
+#include "Coin.h"
+#include "PrivateKey.h"
+#include "Signer.h"
+#include "Base64.h"
+#include "Cosmos/Signer.h"
+#include "Binance/Signer.h"
+#include "Ethereum/Signer.h"
+#include "Ethereum/Address.h"
+#include "proto/Ethereum.pb.h"
+
+#include <google/protobuf/util/json_util.h>
+
+using namespace TW;
+using namespace google::protobuf;
+
+Proto::SigningOutput Signer::sign() const noexcept {
+    const auto coinType = (TWCoinType) input.network();
+    const auto transaction = input.transaction();
+    const auto privateKey = PrivateKey(Data(input.private_key().begin(), input.private_key().end()));
+
+    auto output = Proto::SigningOutput();
+
+    switch (coinType)
+    {
+        case TWCoinTypeCosmos: {
+            Cosmos::Proto::SigningInput message;
+            parse(transaction, &message, output);
+            if (!output.has_error()) {
+                message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+                auto signerOutput = Cosmos::Signer(std::move(message)).build();
+                output.set_json(signerOutput.json());
+            }
+            break;
+        }
+        case TWCoinTypeBinance: {
+            Binance::Proto::SigningInput message;
+            parse(transaction, &message, output);
+            if (!output.has_error()) {
+                message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+                auto signerOutput = Binance::Signer(std::move(message)).build();
+                output.set_encoded(Base64::encode(signerOutput));
```

Comment:
what's the purpose for base64 encoding? it's already encoded as hex

Reply:
I didn't know how it was encoded. Now I converted as string.